### PR TITLE
Fix resource link

### PR DESCRIPTION
**Pull request by ArcturusZhang**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/resourcemanager/Azure.ResourceManager/src/autorest.md [hunk 1]
```diff
@@ -593,6 +589,73 @@ directive:
...
+                    "type": "string",
+                    "description": "The fully qualified ID of the resource link. Use the format, /subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/{provider-namespace}/{resource-type}/{resource-name}/Microsoft.Resources/links/{link-name}. For example, /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myGroup/Microsoft.Web/sites/mySite/Microsoft.Resources/links/myLink",
+                    "x-ms-skip-url-encoding": true
+                },
+                {
+                    "name": "name",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "fuck you"
```
Comment (Yao725) [inline sdk/resourcemanager/Azure.ResourceManager/src/autorest.md:651]: （￣.￣）Remember to change this description

Comment (ArcturusZhang) [reply sdk/resourcemanager/Azure.ResourceManager/src/autorest.md:651]: Oops I forget to change.... Thank you to find out

File: sdk/resourcemanager/Azure.ResourceManager/src/autorest.md [hunk 2]
```diff
@@ -14,9 +14,9 @@ head-as-boolean: false
 mgmt-debug:
   show-request-path: true
 batch:
-  - tag: package-common-type
+#   - tag: package-common-type
```
Comment (Yao725) [inline sdk/resourcemanager/Azure.ResourceManager/src/autorest.md]: Why commented these two tags?

File: sdk/resourcemanager/Azure.ResourceManager/src/autorest.md [hunk 3]
```diff
@@ -114,18 +114,17 @@ output-folder: $(this-folder)/Resources/Generated
...
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/91ac14531f0d05b3d6fcf4a817ea0defde59fe63/specification/resources/resource-manager/Microsoft.Authorization/stable/2020-09-01/dataPolicyManifests.json
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/91ac14531f0d05b3d6fcf4a817ea0defde59fe63/specification/resources/resource-manager/Microsoft.Authorization/stable/2016-09-01/locks.json
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/91ac14531f0d05b3d6fcf4a817ea0defde59fe63/specification/resources/resource-manager/Microsoft.Resources/stable/2016-09-01/links.json
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/91ac14531f0d05b3d6fcf4a817ea0defde59fe63/specification/resources/resource-manager/Microsoft.Resources/stable/2021-01-01/subscriptions.json
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/91ac14531f0d05b3d6fcf4a817ea0defde59fe63/specification/resources/resource-manager/Microsoft.Features/stable/2021-07-01/features.json
+    - C:\Users\dapzhang\Documents\workspace\azure-rest-api-specs/specification/resources/resource-manager/Microsoft.Authorization/stable/2020-09-01/policyAssignments.json
```
Comment (Yao725) [inline sdk/resourcemanager/Azure.ResourceManager/src/autorest.md]: I think we shouldn't change input file.

**Review by Yao725 — APPROVED** [review]

File: sdk/resourcemanager/Azure.ResourceManager/src/autorest.md [hunk 4]
```diff
@@ -593,6 +590,73 @@ directive:
...
+                    "type": "string",
+                    "description": "The fully qualified ID of the resource link. Use the format, /subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/{provider-namespace}/{resource-type}/{resource-name}/Microsoft.Resources/links/{link-name}. For example, /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myGroup/Microsoft.Web/sites/mySite/Microsoft.Resources/links/myLink",
+                    "x-ms-skip-url-encoding": true
+                },
+                {
+                    "name": "name",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "fuck you"
```
Comment (Yao725) [inline sdk/resourcemanager/Azure.ResourceManager/src/autorest.md:621]: It seems that we missed here.

File: sdk/resourcemanager/Azure.ResourceManager/src/autorest.md [hunk 5]
```diff
@@ -593,6 +590,73 @@ directive:
     where: $.definitions.ProviderExtendedLocation.properties.location
     transform: >
       $["x-ms-format"] = "azure-location"
+  # the resource link resource is not using the regular pattern of resources, instead, it is using `/{linkId}` as its creation path
```
Comment (ArthurMa1978) [inline sdk/resourcemanager/Azure.ResourceManager/src/autorest.md:593]: This directive is so big, how about just custom the code?
Is resource link a spark feature that be widely used? If not can suppress it now, and ask service team to fix it from source?

Comment (ArcturusZhang) [reply sdk/resourcemanager/Azure.ResourceManager/src/autorest.md:593]: That would be the best solution.
We will write too much custom code which I do not like it this way

**Comment by ArcturusZhang** [comment]
After an offline discussion, we decide to remove this resource from resourcemanager, and add it back to resources after GA. Closing this PR

**Pull request closed without merging**